packages/xstate-react/src/useMachine.ts

```diff
@@ -115,15 +116,27 @@ export function useMachine<TMachine extends AnyStateMachine>(
     isEqual
   );
 
+  // This timer is used for detecting React 18 strict mode's unmount/remount
+  // double effect execution. It will only stop the service if the component
+  // is truly unmounted.
+  const strictModeTimer = useRef<any>(null);
+
   useEffect(() => {
+    if (strictModeTimer.current) {
+      clearTimeout(strictModeTimer.current);
```

Comment:
Ah, I guess this is already somewhat guarded here:
https://github.com/statelyai/xstate/blob/896f0e207e3db8019083f4d2eac3c99c39a15aec/packages/core/src/interpreter.ts#L520-L523

but I would feel more confident about this if we'd explicitly return here instead of relying on this logic within `start()` ;p